packager/signer/signer.go

```diff
@@ -288,6 +288,41 @@ var promGatewayRequestsTotal = promauto.NewCounterVec(
 	[]string{"code"},
 )
 
+// promGatewayRequestsLatency is a Prometheus summary that observes requests latencies.
+// Objectives key value pairs set target quantiles and respective allowed rank variance.
+// Upon query, for each Objective quantile (0.5, 0.9, 0.99) the summary returns
+// an actual observed latency value that is ranked close to the Objective value.
+// For more intuition on the Objectives see http://alexandrutopliceanu.ro/post/targeted-quantiles/.
+var promGatewayRequestsLatency = promauto.NewSummaryVec(
+	prometheus.SummaryOpts{
+		Name:       "gateway_request_latencies_in_seconds",
+		Help:       "Latencies (in seconds) of gateway requests to AMP document server - by HTTP response status code.",
+		Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
+	},
+	[]string{"code"},
+)
+
+func (this *Signer) fetchURLAndMeasure(fetch *url.URL, serveHTTPReq *http.Request) (*http.Request, *http.Response, *util.HTTPError) {
+	now := time.Now()
```

Comment:
you might want to stub out 'now' so it's easier to inject your own now in tests.

Reply:
Thanks a lot, this is a very cool idea! I didn't do this for top level
requests because latencies are measured by a third-party (Prometheus :),
but it can be done here. 

Done.